hack/test-go.sh

```diff
@@ -54,6 +55,11 @@ go_test_flags=(
 packages=()
 mapfile -t packages < <(go list ./... | grep -v 'sigs.k8s.io/k8s-container-image-promoter/cmd\|test-e2e')
 
-GO111MODULE=on go test "${go_test_flags[@]}" "${packages[@]}"
+for p in "${packages[@]}"; do
+    if ! GO111MODULE=on go test "${go_test_flags[@]}" "${p}"; then
+        # Exit early.
+        exit 1
```

Comment:
As this reads right now, if any suite of package tests fails, the entire script fails.
I think I'd prefer to see this encapsulated in a function, with `return 1` instead.

More below 👇🏾

Reply:
Note that `go_test_flags` includes `-cover -coverprofile "${ARTIFACTS}/coverage.out"`.
By putting this in loop, we'll mangle the results of `go tool cover`. Now it will only include the coverage report from the last package that gets tested in the loop.